app/api/probe_dock/metrics_api.rb

```diff
@@ -172,6 +178,48 @@ def current_organization
         }
       end
 
+      get :testsByWeek do
+        authorize! :organization, :data
+
+        nb_weeks = params[:nbWeeks].to_i
+        nb_weeks = MAX_NB_WEEKS_FOR_TESTS if nb_weeks > MAX_NB_WEEKS_FOR_TESTS
+        nb_weeks = DEFAULT_NB_WEEKS_FOR_TESTS if nb_weeks <= 0
+        nb_weeks
+
+        start_date = (nb_weeks - 1).weeks.ago.beginning_of_week
+
+        rel = ProjectTest.joins(:project).where('projects.organization_id = ?', current_organization.id).select('count(project_tests.id) as total_tests_count')
+        rel = rel.where 'project_tests.first_run_at < ?', start_date
+
+        if params[:projectIds].present? && params[:projectIds].kind_of?(Array)
+         rel = rel.where 'projects.api_id IN (?)', params[:projectIds].collect(&:to_s)
+        end
+
+        if params[:userIds].present? && params[:userIds].kind_of?(Array)
+         users = User.where('api_id IN (?)', params[:userIds].collect(&:to_s)).to_a
+         rel = rel.joins 'LEFT OUTER JOIN test_keys ON project_tests.key_id = test_keys.id'
+
+         user_ids = users.collect &:id
+         rel = rel.where '(test_keys.user_id IS NOT NULL AND test_keys.user_id IN (?)) OR (test_keys.user_id IS NULL AND project_tests.first_runner_id IN (?))', user_ids, user_ids
+        end
```

Comment:
Extract the two ifs above into a `filter_new_tests_rel` method that can be reused here and in `count_new_tests_by_interval`.